### PR TITLE
Fixing #1171: BleiCorpus after serialize cannot be loaded

### PR DESCRIPTION
**Pull request by pranaydeep-af**
@tmylk. Apologies for the delay. Had to deal with some college course work.
I went through the save() and load() methods and did a few experiments. The save() method stores a pickle dump of the particular object. So in the case of a corpora.dictionary object that is rendered useless due to the presence of a superior method ie. save_corpus(), however the function may be handy when used for saving pickles of some other objects.
So I have added a check to see if the function was called from a corpora object and if it was, I have raised an exception. Otherwise the function works as usual.
For the load function I have added a warning to not use it for loading corpora objects.
Do tell me if some naive mistake has been made.

### CODE REVIEW

**Comment by tmylk** [comment]
Travis tests re-ran after smart_open update

File: gensim/utils.py [hunk 1]
```diff
@@ -830,7 +844,7 @@ def run(self):
         it = iter(self.corpus)
         while True:
             chunk = itertools.islice(it, self.chunksize)
-            if self.as_numpy:
+            if self.as_numpy:
```
Comment (piskvorky) [inline gensim/utils.py]: PEP8: no trailing whitespace.

File: gensim/utils.py [hunk 2]
```diff
@@ -490,6 +497,13 @@ def save(self, fname_or_handle, separately=None, sep_limit=10 * 1024**2,
         in both Python 2 and 3.
 
         """
+        import os
```
Comment (piskvorky) [inline gensim/utils.py]: Imports belong to the top of file (separated into builtin - external - internal blocks).

(here and elsewhere)

File: gensim/utils.py [hunk 3]
```diff
@@ -260,6 +260,9 @@ def load(cls, fname, mmap=None):
         is encountered.
 
         """
+        import warnings
+        warnings.warn("This method loads pickle dumps. If a corpora was saved with the save_corpus() method," +
```
Comment (piskvorky) [inline gensim/utils.py]: The message, when emitted, will leave the user confused -- what method?

`corpora` is plural of `corpus`. Do not split messages in the middle with `\n`, leave formatting to the terminal. End sentences in full stop.

File: gensim/utils.py [hunk 4]
```diff
@@ -358,6 +361,10 @@ def _smart_save(self, fname, separately=None, sep_limit=10 * 1024**2,
         in both Python 2 and 3.
 
         """
+        import os
+        directory = os.path.dirname(fname)
+        if not os.path.exists(directory):
```
Comment (piskvorky) [inline gensim/utils.py]: Definitely not -- no creating of directories please. Non-existent path should end in an exception (as it was before).

File: gensim/utils.py [hunk 5]
```diff
@@ -490,6 +497,13 @@ def save(self, fname_or_handle, separately=None, sep_limit=10 * 1024**2,
         in both Python 2 and 3.
 
         """
+        import os
+        directory = os.path.dirname(fname_or_handle)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        import corpora.dictionary
+        if isinstance(self, corpora.dictionary.Dictionary):
+            raise TypeError('The method is not built to handle corpora. Use the save_corpus() method')
```
Comment (piskvorky) [inline gensim/utils.py]: The `save` method can handle corpora fine, it's just probably not what the user wants. Plus, you're testing here for `Dictionary` anyway, not corpora.

The recommended way for serializing corpora is `SomeCorpusClass.serialize()`, not `save_corpus()`.

**Comment by tmylk** [comment]
Closing as abandoned.
Also this needs to be specced out better as to what exactly the fix should be.

**Pull request closed without merging**